### ISSUE TITLE
Refactor `capstone.utils.git` module

### ISSUE THREAD

**Issue by nikochiko** [pipalacademy/capstone#64]
There are a few problems with it:
- It has raw commands only and each command accepts a "workdir" argument and then runs git as a subprocess. This allows some room for mistake from the developer if they forget to mention it and end up making changes to the current working directory (which would almost always be the Git directory of the capstone project).
- It manually does a `cd {workdir}; git ...`. This is bad because if the `cd ...` step fails, it won't be caught and bash will continue on to the next `git` command wherever it is. This could again make changes to the project's directory. Methods in the `subprocess` library accept a `cwd` argument. This should be used instead.

Here's what I have in mind:
- Restructure the functions to use `cwd` argument instead of manually doing `cd ...`.
- Make a class `Repo` that can be initialised like: `Repo(path)` and then add these same functions as methods on it. This is much easier to work with.

**Comment by nikochiko** [comment]
I had initially written this module with the objective of keeping it very transparent. Anything passed to the methods will be directly interpreted as either CLI options or args.
But this is turning out to be a bad abstraction. I spotted two instances in code where workdir was being used in `git.clone(...)`, when the destination directory didn't exist yet. `cd ...` failed but the command gave the wrong impression that the working directory had been changed.
In some cases, it makes sense to specially handle args/kwargs rather than directly passing them to CLI. For example,
- `git.clone(url, dest=repo_dir)` is clearer than `git.clone(url, repo_dir)`
- `git.commit("initial commit")` is straightforward but it must be written as `git.commit(message="initial commit")`
